### DISCUSSION TITLE
noRestrictedGlobals - allow adding a message

### DISCUSSION THREAD

**Discussion by osdiab** [biomejs/biome#4721] [Ideas]
Right now the rule doesn't allow explaining why a global is restricted, which makes it worse than the ESLint equivalent; this is important for our process because it allows us to communicate to the team why we are avoiding certain globals and what you should do instead. Would like the same thing for the Biome equivalent rule!

**Comment by ematipico** [comment]
What prevents you from adding a comment in the options? 

```json5
// biome.jsonc
{
    "options": {
		// someGlobal shouldn't be used because... 
        "deniedGlobals": ["$", "someGlobal"]
    }
}
```

**Reply by dyc3** [reply]
This requires opening another file, and with monorepo support/extended configs, it might not be clear which config file to open. Allowing users to add a message in the diagnostic itself avoids this entirely.

**Reply by osdiab** [reply]
Yes, I can guarantee you that with a larger team, very few people will search through a large biome configuration when their IDE tells them something is going wrong, and they themselves have never touched the biome config before. My team currently does this with ESLint and the errors telling you what to do as you’re doing it is extremely helpful.

**Reply by dyc3** [reply]
I've opened #4722 for this.
